src/app/content/utils/canonicalUrl.ts

```diff
@@ -0,0 +1,40 @@
+import { CANONICAL_MAP } from '../../../canonicalBookMap';
+import { BOOKS } from '../../../config';
+import { AppServices } from '../../types';
+import { assertDefined } from '../../utils';
+import { formatBookData } from '../utils';
+import { findArchiveTreeNode } from './archiveTreeUtils';
+import { getUrlParamForPageTitle } from './urlUtils';
+
+export async function getBook(
```

Comment:
on my other branch i added a function that does the same thing as this, https://github.com/openstax/rex-web/pull/258/files#diff-72e23ff60301d48e0ffae98f76dd57b4R30 content/utils is probably a better home for a generic book loader rather than `canonicalUrl` helper